src/main/scala/org/hammerlab/guacamole/ReadSet.scala

```diff
@@ -53,8 +53,8 @@ case class ReadSet(
 
   lazy val mappedPairedReads: RDD[PairedRead[MappedRead]] = reads.flatMap(read =>
     read match {
-      case rp: PairedRead[MappedRead] => Some(rp)
-      case _                          => None
+      case rp if rp.isMapped => Some(rp.asInstanceOf[PairedRead[MappedRead]])
```

Comment:
This doesn't check that `rp` is a `ReadPair` though, it may just be a `MappedRead`